### ISSUE TITLE
Local dependencies lead to build panic

### ISSUE THREAD

**Issue by KoltesDigital** [bytecodealliance/wasm-tools#937]
Hi,

I'm trying to update _crates/wasm-compose/example_ to use current version of `cargo component`.

When I declare `middleware`'s dependency to `service`:

```toml
[package.metadata.component.dependencies]
	service = { path = "../service/target/wasm32-unknown-unknown/release/service.wasm" }
```

a wild compilation panic: appears:

    thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', C:\Users\X\.cargo\registry\src\github.com-1ecc6299db9ec823\wit-component-0.6.0\src\encoding\wit.rs:212:24

https://github.com/bytecodealliance/wasm-tools/blob/8d16aad76fd1e2995c28bf4ec97cfb634bd2a60a/crates/wit-component/src/encoding/wit.rs#L218

I barely understand that dependencies are expected to have an URL, which local ones do not have. Or am I wrong?

Note that the `middleware` WIT does not import `service` because it's still unclear to me and I can't get it to work neither. If I add:

    import service: service.component

then build fails with that error instead:

```
no package dependency specified for `service`
       --> X\wasm-tools\crates\wasm-compose\example\middleware\./middleware.wit:16:21
        |
     16 |     import service: service.component
        |                     ^------
```

In brief, can I please have some guidance on compiling a component with local dependencies? I'll gladly PR the changes once they work.

**Comment by peterhuene** [comment]
Hi @KoltesDigital. Sorry you ran into this and thank you for the reminder to update this example.

I think this is a bug in `cargo-component`. Let me take a closer look shortly, although I'm mostly off this week.

**Comment by peterhuene** [comment]
Hi @KoltesDigital.

I've put up a PR for fixing the example with the latest `cargo-component` (caveat: it does require bytecodealliance/cargo-component#52 merged to work).

I'll be tracking down the panic issue with using the service component as a direct dependency (as you have it above) too.

**Comment by KoltesDigital** [comment]
@peterhuene Thanks! Obviously I would never have been able to do that :)

I installed the mentioned rev of `cargo component`, and successfully compiled `service` and `middleware`. However `server` is failing to compile:

```
error: the wasm32-unknown-unknown target is not supported by default, you may need to enable the "js" feature. For more information see: https://docs.rs/getrandom/#webassembly-support
   --> C:\Users\X\.cargo\registry\src\github.com-1ecc6299db9ec823\getrandom-0.2.8\src\lib.rs:263:9
    |
263 | /         compile_error!("the wasm32-unknown-unknown target is not supported by \
264 | |                         default, you may need to enable the \"js\" feature. \
265 | |                         For more information see: \
266 | |                         https://docs.rs/getrandom/#webassembly-support");
    | |________________________________________________________________________^

error[E0433]: failed to resolve: use of undeclared crate or module `imp`
   --> C:\Users\X\.cargo\registry\src\github.com-1ecc6299db9ec823\getrandom-0.2.8\src\lib.rs:290:5
    |
290 |     imp::getrandom_inner(dest)
    |     ^^^ use of undeclared crate or module `imp`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `getrandom` due to 2 previous errors
```

`getrandom` is not a direct dependency and there are even two versions of it. However something tells me not to add the `js` feature anyway. Can you please document how you've been able to compile it?

**Comment by peterhuene** [comment]
Hi @KoltesDigital.

For the server, make sure you're not building it with `cargo component build` as it is intended to build natively for your platform.

The docs just use `cargo run`: `cargo run --release -- ../service/target/wasm32-unknown-unknown/release/svc.wasm` to build and run it.

Let me know if that isn't the problem and I can look into it.

**Comment by KoltesDigital** [comment]
Agreed, I was too quick to press Up Enter whereas it totally makes sense that the server is a regular binary. That also explains the use of `wasmtime_component_macro::bindgen!`.

Thanks for your efforts!
